VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache.service.ts

```diff
@@ -114,16 +114,38 @@ export class VideoControlCacheService {
             this.logger.warn(`${this.loggerPrefix} Cannot set hand raise status as hearing control states is not initialised.`);
             return;
         }
-
-        if (!this.hearingControlStates.participantStates[participantId]) {
-            this.hearingControlStates.participantStates[participantId] = { isHandRaised: isHandRaisedValue };
-        } else {
-            this.hearingControlStates.participantStates[participantId].isHandRaised = isHandRaisedValue;
-        }
-
-        if (syncChanges) {
-            this.savingHearingState();
-        }
+        const self = this;
+        this.storageService
+            .loadHearingStateForConference(this.conferenceId)
+            .pipe(take(1))
+            .toPromise()
+            .then(state => {
+                self.hearingControlStates = state;
+                self.logger.info(`${this.loggerPrefix} re-initialised state for ${this.conferenceId}.`, {
+                    hearingControlStates: this.hearingControlStates
+                });
+                raiseHandInCache();
+            })
+            .catch(() => {
+                self.logger.info(
+                    `${this.loggerPrefix} failed to re-initialised state for ${this.conferenceId}.
+                Control States may be out of sync`,
+                    {
+                        hearingControlStates: this.hearingControlStates
+                    }
+                );
+                raiseHandInCache();
+            });
+        const raiseHandInCache = () => {
```

Comment:
Just a thought - if this can raise or lower the hand, maybe name it `setRaiseHandStatusInCache` just so it's clear that it can do both?